.github/workflows/ci.yaml

```diff
@@ -0,0 +1,133 @@
+---
+name: "CI"
+on:
+  push:
+    branches: ["develop"]
+  pull_request:
+    branches: ["develop"]
+jobs:
+  Linting:
+    runs-on: "ubuntu-latest"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+    steps:
+      # ----------------------------------------------
+      #       check-out repo and set-up python
+      # ----------------------------------------------
+      - uses: "actions/checkout@v2"
+      - uses: "actions/setup-python@v2"
+        with:
+          python-version: 3.9
+      # ----------------------------------------------
+      #        load pip cache if cache exists
+      # ----------------------------------------------
+      - uses: "actions/cache@v2"
+        with:
+          path: "~/.cache/pip"
+          key: "${{ runner.os }}-pip"
+          restore-keys: "${{ runner.os }}-pip"
+      # ----------------------------------------------
+      #          install and run linters
+      # ----------------------------------------------
+      - run: "python -m pip install --upgrade pip wheel"
+      - run: "python -m pip install invoke black=='20.8b1' flake8=='3.9.2'"  # Until there is a poetry install --dev-only (poetry 1.2) install the linting tools here to speed up this step
```

Comment:
Any way to pull the specific versions from pyproject.toml? I can see this easily getting out of sync by accident otherwise.

Reply:
we could try to do something interesting/hacky here, but I think the best bet is to wait for poetry 1.2 and then do `poetry install --dev-only`